service/device/registry/internal/src/main/java/org/eclipse/kapua/service/device/registry/connection/option/internal/DeviceConnectionOptionServiceImpl.java

```diff
@@ -62,9 +64,16 @@ public DeviceConnectionOption update(DeviceConnectionOption deviceConnectionOpti
         //
         // Check Access
         KapuaLocator locator = KapuaLocator.getInstance();
+        DeviceConnectionOptionQuery query = new DeviceConnectionOptionQueryImpl(deviceConnectionOptions.getScopeId());
         AuthorizationService authorizationService = locator.getService(AuthorizationService.class);
         PermissionFactory permissionFactory = locator.getFactory(PermissionFactory.class);
         authorizationService.checkPermission(permissionFactory.newPermission(DeviceDomains.DEVICE_CONNECTION_DOMAIN, Actions.write, deviceConnectionOptions.getScopeId()));
+        DeviceConnectionOptionListResult list = query(query);
+        for (DeviceConnectionOption deviceConnectionOptionDB : list.getItems()) {
```

Comment:
Instead of asking for all the reservation for a given scope and then iterate through all of them, I would rather look for the specific association (i.e. the one with the same user id but a different client id): if at least one exists you are already sure you need to throw the exception. The result is the same of your proposal, but it's better optimized since you don't have to iterate and check multiple useless items.